src/background/track.js

```diff
@@ -41,24 +42,70 @@ export const startTracking = (sessionId, originalWindowId, openedWindowId) => {
   // 同一のトラッキングセッションが複数開かれた際に、最後に開かれたもののみ追跡する
   trackingWindows = trackingWindows.filter(x => x.openedWindowId != originalWindowId && x.originalWindowId != originalWindowId);
   trackingWindows.push({ sessionId, originalWindowId, openedWindowId });
+  setFocusedId(openedWindowId);
 
   if (!IsTracking) {
+    browser.windows.onFocusChanged.addListener(setFocusedId);
     browser.windows.onRemoved.addListener(endTrackingByWindowClose);
     browser.windows.onCreated.addListener(handleCreateWindow);
     IsTracking = true;
   }
 
   updateTrackingStatus();
-  log.info(logDir, "startTracking()", trackingWindows);
+  log.info(logDir, "startTracking()", sessionId, originalWindowId, openedWindowId, trackingWindows);
 };
 
+export const setFocusedId = (windowId) => {
+  lastWindowId = windowId;
+  log.info(logDir, "setFocusedId()", windowId);
+}
+
 const handleCreateWindow = (window) => {
-  if (getSettings("shouldTrackNewWindow")) startTracking(trackingWindows[trackingWindows.length - 1].sessionId, window.id, window.id);
+  const trackedWindowId = lastWindowId;
+  if (!getSettings("shouldTrackNewWindow")) return;
+
+  // If the last window wasn't a valid window, skip
+  if (trackedWindowId < 0) return;
+
+  const focusedWindow = trackingWindows.find(x => x.openedWindowId === trackedWindowId || x.originalWindowId === trackedWindowId);
+  // If the last window that we were in isn't being tracked, don't track the new window
+  if (!focusedWindow) return;
```

Comment:
**BUG FIX** for #1186: Prevents a tracked session from hijacking another session's window.